File: frontend/src/types/vehiculo.ts
export type EstadoVehiculo =
  | 'DISPONIBLE'
  | 'EN_TALLER'
  | 'EN_REPARACION'
  | 'LISTO'
  | 'ENTREGADO';

export interface Vehiculo {
  id: number;
  clienteId: number;
  clienteNombre?: string;
  placa: string;
  marca: string;
  modelo: string;
  anio?: number;
  color?: string;
  kilometraje: number;
  estado: EstadoVehiculo;
  estadoLabel: string;
  notas?: string;
  fechaCreacion: string;
  fechaActualizacion: string;
}

export interface VehiculoRequest {
  placa: string;
  marca: string;
  modelo: string;
  anio?: number;
  color?: string;
  kilometraje?: number;
  notas?: string;
  estado?: EstadoVehiculo;
  /** Solo lo envia el staff al crear/reasignar vehiculo de un cliente. */
  clienteId?: number;
}

File: backend/voltios_y_ruedas/src/main/java/com/voltiosyruedas/taller/vehiculo/entity/Vehiculo.java
package com.voltiosyruedas.taller.vehiculo.entity;

import com.voltiosyruedas.taller.auth.entity.Usuario;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "vehiculos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vehiculo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Usuario cliente;

    @Column(nullable = false, unique = true, length = 20)
    private String placa;

    @Column(nullable = false, length = 80)
    private String marca;

    @Column(nullable = false, length = 80)
    private String modelo;

    @Column(name = "anio")
    private Integer anio;

    @Column(length = 40)
    private String color;

    @Column(nullable = false)
    @Builder.Default
    private Integer kilometraje = 0;

    @Column(nullable = false, length = 50)
    @Builder.Default
    private String estado = "DISPONIBLE";

    @Column(columnDefinition = "TEXT")
    private String notas;

    @Column(name = "fecha_creacion", nullable = false, updatable = false)
    private LocalDateTime fechaCreacion;

    @Column(name = "fecha_actualizacion", nullable = false)
    private LocalDateTime fechaActualizacion;

    @PrePersist
    protected void onCreate() {
        fechaCreacion = LocalDateTime.now();
        fechaActualizacion = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        fechaActualizacion = LocalDateTime.now();
    }

    public enum EstadoVehiculo {
        DISPONIBLE,
        EN_TALLER,
        EN_REPARACION,
        LISTO,
        ENTREGADO
    }
}
